src/Internal/Service/Blockfrost.purs

```diff
@@ -0,0 +1,95 @@
+module Ctl.Internal.Service.Blockfrost
+  ( BlockfrostServiceM
+  , BlockfrostServiceParams
+  , runBlockfrostServiceM
+  ) where
+
+import Prelude
+
+import Aeson (class DecodeAeson, decodeAeson, parseJsonStringToAeson)
+import Affjax (Error, Response, URL, defaultRequest, request) as Affjax
+import Affjax.RequestHeader (RequestHeader(RequestHeader)) as Affjax
+import Affjax.ResponseFormat (string) as Affjax.ResponseFormat
+import Affjax.StatusCode (StatusCode(StatusCode)) as Affjax
+import Control.Monad.Reader.Class (ask)
+import Control.Monad.Reader.Trans (ReaderT, runReaderT)
+import Ctl.Internal.Contract.QueryBackend (BlockfrostBackend)
+import Ctl.Internal.ServerConfig (ServerConfig, mkHttpUrl)
+import Ctl.Internal.Service.Error
+  ( ClientError(ClientHttpError, ClientHttpResponseError, ClientDecodeJsonError)
+  , ServiceError(ServiceBlockfrostError)
+  )
+import Data.Bifunctor (lmap)
+import Data.Either (Either(Left, Right))
+import Data.HTTP.Method (Method(GET))
+import Data.Maybe (Maybe, maybe)
+import Effect.Aff (Aff)
+import Effect.Aff.Class (liftAff)
+import Undefined (undefined)
+
+--------------------------------------------------------------------------------
+-- BlockfrostServiceM
+--------------------------------------------------------------------------------
+
+type BlockfrostServiceParams =
+  { blockfrostConfig :: ServerConfig
+  , blockfrostApiKey :: Maybe String
+  }
+
+type BlockfrostServiceM (a :: Type) = ReaderT BlockfrostServiceParams Aff a
+
+runBlockfrostServiceM
+  :: forall (a :: Type). BlockfrostBackend -> BlockfrostServiceM a -> Aff a
+runBlockfrostServiceM backend = flip runReaderT serviceParams
+  where
+  serviceParams :: BlockfrostServiceParams
+  serviceParams =
+    { blockfrostConfig: backend.blockfrostConfig
+    , blockfrostApiKey: backend.blockfrostApiKey
+    }
+
+--------------------------------------------------------------------------------
+-- Making requests to Blockfrost endpoints
+--------------------------------------------------------------------------------
+
+data BlockfrostEndpoint
+
+realizeEndpoint :: BlockfrostEndpoint -> Affjax.URL
+realizeEndpoint endpoint =
+  case endpoint of
+    _ -> undefined
+
+blockfrostGetRequest
+  :: BlockfrostEndpoint
+  -> BlockfrostServiceM (Either Affjax.Error (Affjax.Response String))
+blockfrostGetRequest endpoint = ask >>= \params -> liftAff do
+  Affjax.request $ Affjax.defaultRequest
+    { method = Left GET
+    , url = mkHttpUrl params.blockfrostConfig <> realizeEndpoint endpoint
+    , responseFormat = Affjax.ResponseFormat.string
+    , headers =
+        maybe mempty (\apiKey -> [ Affjax.RequestHeader "project_id" apiKey ])
+          params.blockfrostApiKey
+    }
+
```

Comment:
Can we add this while we're at it?

```suggestion
blockfrostPostRequest
  :: BlockfrostEndpoint
  -> MediaType
  -> Maybe RequestBody
  -> BlockfrostServiceM (Either Affjax.Error (Affjax.Response String))
blockfrostPostRequest endpoint mediaType mbContent = ask >>= \params -> liftAff do
  Affjax.request $ Affjax.defaultRequest
    { method = Left POST
    , url = mkHttpUrl params.blockfrostConfig <> realizeEndpoint endpoint
    , content = mbContent
    , responseFormat = Affjax.ResponseFormat.string
    , headers =
        [ Affjax.ContentType mediaType ] <>
        maybe mempty (\apiKey -> [ Affjax.RequestHeader "project_id" apiKey ])
          params.blockfrostApiKey
    }
```

Reply:
Actually I guess it could be `Maybe (MediaType /\ RequestBody)`, not critical though